app/decorators/endpoint_decorator.rb

```diff
@@ -0,0 +1,24 @@
+class EndpointDecorator < ApplicationDecorator
+  delegate_all
+
+  def cas_usage_icon(cas_usage_name)
+    cas_usage_to_icon(cas_usage_name) ||
+      cas_usage_to_icon_optionnal(cas_usage_name) ||
+      cas_usage_to_icon_forbidden(cas_usage_name) ||
+      '-'
+  end
+
+  private
+
+  def cas_usage_to_icon(cas_usage_name)
+    '<span style="color:#1f8d49" class="fr-icon-checkbox-circle-fill" aria-hidden="true"></span>' if use_case?(cas_usage_name)
+  end
+
+  def cas_usage_to_icon_optionnal(cas_usage_name)
+    '<span style="color:#1f8d49" class="fr-icon-checkbox-circle-line" aria-hidden="true"></span>' if use_case_optionnal?(cas_usage_name)
+  end
+
+  def cas_usage_to_icon_forbidden(cas_usage_name)
+    '<span style="color:#d64d00" class="fr-icon-close-circle-line" aria-hidden="true"></span>' if use_case_forbidden?(cas_usage_name)
+  end
```

Comment:
Ça peut encore se refactor ici:

```suggestion
  def cas_usage_to_icon(cas_usage_name)
    cas_usge_icon_tag('#1f8d49')if use_case?(cas_usage_name)
  end

  def cas_usage_to_icon_optionnal(cas_usage_name)
    cas_usge_icon_tag('#1f8d49') if use_case_optionnal?(cas_usage_name)
  end

  def cas_usage_to_icon_forbidden(cas_usage_name)
    cas_usge_icon_tag('#d64d00') if use_case_forbidden?(cas_usage_name)
  end
  
  def cas_usage_icon_tag(color)
    "<span style=\"color:#{color}\" class=\"fr-icon-close-circle-line\" aria-hidden=\"true\"></span>"
  end
```